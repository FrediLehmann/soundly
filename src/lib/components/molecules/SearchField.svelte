<script type="ts">
  import { Search } from '$lib/components/Icons';
  import { Button, Input } from '$lib/components/atoms';
</script>

<div class="flex w-full">
  <Input
    name="Search"
    label="Search"
    value=""
    labelSrOnly={true}
    placeholder="Search"
    class="w-full py-0" />
  <Button class="px-2">
    <Search />
  </Button>
</div>
