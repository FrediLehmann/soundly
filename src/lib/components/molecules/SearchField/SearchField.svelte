<script type="ts" context="module">
  import { Search } from '$lib/Icons';
</script>

<div>
  <input type="text" name="Search" aria-label="Search" placeholder="Search" />
  <button aria-label="Search">
    <Search stroke-width="3" width="1.3rem" height="1.3rem" />
  </button>
</div>

<style>
  div {
    display: flex;
    width: 100%;
  }

  input {
    outline: none;
    padding: 0.25rem 1rem;
    width: 100%;

    font-size: 1.5rem;

    color: hsla(var(--neutral-700));
    background-color: hsla(var(--neutral-100));

    border: 1px solid hsla(var(--neutral-300));
    border-start-start-radius: 0.125rem;
    border-end-start-radius: 0.125rem;
  }

  input:focus,
  input:active {
    box-shadow: 0 0 0 1px hsla(var(--primary-400));
  }

  button {
    cursor: pointer;
    background: hsla(var(--primary))
      linear-gradient(
        20deg,
        hsla(var(--primary), 0.6) 20%,
        hsla(var(--secondary), 0.6) 80%
      );

    color: hsla(var(--neutral-000));

    padding-inline: 1rem;

    border: none;
    border-start-end-radius: 0.125rem;
    border-end-end-radius: 0.125rem;
  }

  button:hover {
    box-shadow: 0 0 0 2px hsla(var(--primary-400));
  }

  button:focus,
  button:active {
    box-shadow: 0 0 0 1px hsla(var(--neutral-000), 50%),
      0 0 0 3px hsla(var(--primary-400)), 0 0 #0000;
  }
</style>
