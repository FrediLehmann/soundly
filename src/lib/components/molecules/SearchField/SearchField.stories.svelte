<script>
  import { Meta, Story } from '@storybook/addon-svelte-csf';
  import { userEvent, within } from '@storybook/testing-library';

  import SearchField from './SearchField.svelte';
</script>

<Meta title="Molecules/SearchField" />

<Story
  name="SearchField"
  play={async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const button = canvas.getByRole('button');
    userEvent.hover(button);

    const input = canvas.getByRole('textbox');
    userEvent.hover(input);

    userEvent.type(input, 'Search content');
  }}>
  <SearchField />
</Story>
