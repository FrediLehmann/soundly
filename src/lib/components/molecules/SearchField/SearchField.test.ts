/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import SearchField from './SearchField.svelte';

describe('<SearchField /> component', () => {
  test('renders correctly', () => {
    const { component } = render(SearchField);
    expect(component).toMatchSnapshot();
  });
});
