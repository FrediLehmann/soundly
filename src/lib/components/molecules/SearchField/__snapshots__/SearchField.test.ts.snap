// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> component renders correctly 1`] = `
SearchField {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {},
    "root": <div>
      <div
        class="svelte-1dwmca1"
      >
        <input
          aria-label="Search"
          class="svelte-1dwmca1"
          name="Search"
          placeholder="Search"
          type="text"
        />
         
        <button
          aria-label="Search"
          class="svelte-1dwmca1"
        >
          <svg
            fill="none"
            height="1.3rem"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="3"
            viewBox="0 0 24 24"
            width="1.3rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </button>
      </div>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$capture_state": [Function],
}
`;
